lua/lspconfig/efm.lua

```diff
@@ -18,7 +20,7 @@ https://github.com/mattn/efm-langserver
 General purpose Language Server that can use specified error message format generated from specified command.
 ]];
     default_config = {
-      root_dir = [[root_pattern(".git")]];
+      root_dir = [[root_pattern(".git") or dirname]];
```

Comment:
I would make this `util.root_pattern(".git")(fname) or util.path.dirname(fname)`

Reply:
I followed the same pattern the docs for other have here.

Reply:
Yes, they are wrong :)

Reply:
Why not include the whole block though?

```lua
    root_dir = function(fname)
      return util.root_pattern(".git")(fname) or util.path.dirname(fname)
    end;
```

I don't think the snippet above can be copy-pasted anyway due to the function taking a param, right?

Reply:
It's just text, in fact, if you leave out root_dir entirely that's exactly what will be pasted by docgen.